controllers/controller_test.go

```diff
@@ -542,3 +651,44 @@ func newEtcdMachine(etcdadmCluster *etcdv1.EtcdadmCluster, cluster *clusterv1.Cl
 		},
 	}
 }
+
+func setUpEtcdTest(t *testing.T) (*etcdadmClusterTest, client.WithWatch, *mocks.MockEtcdClient, *http.Client) {
```

Comment:
IMO this name is to generic for what this method does so it can be misleading

maybe something like?
```suggestion
// setupEtcdScalingTest scaffolds resources, clients and mocks to test EtcdCluster scaling.
func setupEtcdScalingTest(t *testing.T) (*etcdadmClusterTest, client.WithWatch, *mocks.MockEtcdClient, *http.Client) {
```